### ISSUE TITLE
facebook widget isn't showing up on mobile

### ISSUE THREAD

**Issue by skippy** [sjifire/website#119]

**Comment by icegulch** [comment]
Appears to be Safari-only? Shows in Chrome and FF

**Comment by skippy** [comment]
yeah, safari doesn't show up, desktop or mobile.  I can't figure it out.... and dropping down to FB javascript just injects a bunch of js and then an iframe.  This iframe shows up but only to a height of 71px...

**Comment by icegulch** [comment]
After some sleuthing, it appears the problem could be solved by using Facebook's Javascript SDK. We're currently using the IFRAME version of the "page plugin":

https://developers.facebook.com/docs/plugins/page-plugin

The trick with the Javascript SDK is you need to include a Facebbok "app ID", which as I recall is tied to one's personal Facebook ID. Maybe that policy has changed? 

I'm reluctant to use my Facebook account -- I only login to see photos of my nieces and nephews. :) 

Would you consider using yours, Adam?

**Comment by icegulch** [comment]
Oops...looks like you tried that. Sorry! I can confirm that it's something about Safari's x-frame-options setting.

**Comment by skippy** [comment]
@icegulch how close are we to this?  I would say *IF* it is an easy win (< 30 min) lets wrap it up.  If not, I'll close it

**Comment by icegulch** [comment]
hi, @skippy. sorry not to be clearer. the issue is well known with respect to Facebook's "page plugin". i don't think it's a ~30min fix from a dev standpoint. 

apparently, it's an ios 15+ issue, which can be "fixed" if a user changes their security settings (if they uncheck "Hide IP from Tracker") -- basically, Safari is trying to prevent Facebook from tracking users, even from embedded widgets.

So, either the user must uncheck that setting in their Safari Settings or not see this (or any other FB widget on any other site).

What I was speculating at above was that things might be different if you use FB's Javascript SDK to embed the widget. I'm not 100% sure of that, though. With the SDK, I think you have to use your personal Facebook "app ID" to power the widget.

I'm not a big fan of having a FB account just to embed their widget.

<img width="772" alt="Screenshot 2023-02-21 at 14 12 44" src="https://user-images.githubusercontent.com/794708/220461013-22b9cd02-533d-485f-89fb-a9434bbfba24.png">

**Comment by icegulch** [comment]
hi, @skippy. sorry not to be clearer. the issue is well known with respect to Facebook's "page plugin". i don't think it's a ~30min fix from a dev standpoint. 

apparently, it's an ios 15+ issue, which can be "fixed" if a user changes their security settings (if they uncheck "Hide IP from Tracker") -- basically, Safari is trying to prevent Facebook from tracking users, even from embedded widgets.

So, either the user must uncheck that setting in their Safari Settings or not see this (or any other FB widget on any other site).

What I was speculating at above was that things might be different if you use FB's Javascript SDK to embed the widget. I'm not 100% sure of that, though. With the SDK, I think you have to use your personal Facebook "app ID" to power the widget.

I'm not a big fan of having a FB account just to embed their widget.

<img width="772" alt="Screenshot 2023-02-21 at 14 12 44" src="https://user-images.githubusercontent.com/794708/220461013-22b9cd02-533d-485f-89fb-a9434bbfba24.png">
